src/components/Reports/ReportsTrafficSources/ReportsTrafficSources.js

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect, useState } from 'react';
+import { InjectAppServices } from '../../../services/pure-di';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import Loading from '../../Loading/Loading';
+
+const ReportsTrafficSources = function({
+  domainName,
+  dateFrom,
+  intl,
+  dependencies: { datahubClient },
+}) {
+  const [trafficSources, setTrafficSources] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [totalQuantity, setTotalQuantity] = useState(0);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setIsLoading(true);
+      const trafficSourcesData = await datahubClient.getTrafficSourcesByPeriod(
+        domainName,
+        dateFrom,
+      );
+      if (trafficSourcesData.success && trafficSourcesData.value) {
+        setTrafficSources(trafficSourcesData.value);
+        const total = Object.keys(trafficSources).reduce(function(previous, key) {
+          return previous + trafficSources[key].quantity;
+        }, 0);
+        setTotalQuantity(total);
+      }
+      setIsLoading(false);
+    };
+
+    fetchData();
+  }, [datahubClient, dateFrom, domainName, trafficSources]);
+
+  return (
+    <div className="wrapper-ranking">
+      <div className="reports-box">
+        <h4 className="title-ranking">
+          <FormattedMessage id="trafficSources.title" />
+        </h4>
+        <div>
+          {isLoading ? (
+            <Loading />
+          ) : trafficSources ? (
+            trafficSources.map((trafficSource, index) => (
+              <div key={index}>
+                <span>{trafficSource.sourceName}</span>
+                <span>{trafficSource.quantity}</span>
+                <span>({((trafficSource.quantity * 100) / totalQuantity).toFixed(2)}%)</span>
```

Comment:
⚠️  I think that you should use localization utilities here.

Reply:
i changed